packages/ember-views/lib/streams/key_stream.js

```diff
@@ -38,6 +38,10 @@ merge(KeyStream.prototype, {
     if (nextObj !== prevObj) {
       if (prevObj && typeof prevObj === 'object') {
         removeObserver(prevObj, this.key, this, this.notify);
+
+        if (nextObj && typeof nextObj === 'object') {
```

Comment:
This should be moved out of the conditional (to handle the case when `prevObj` is `null`).